students/Nick_Lenssen/lesson09/donor_models.py

```diff
@@ -0,0 +1,106 @@
+import pathlib
+
+class Donor:
+
+    def __init__(self, full_name):
+        self.name = full_name
+        self.donations = []
+
+    @property
+    def total_donation(self):
+        """sum of donation history property"""
+        return sum(self.donations)
+
+    @property
+    def num_donation(self):
+        """number of donations with a donors history"""
+        return len(self.donations)
+
+    @property
+    def ave_donation(self):
+        """average donation of a donor's history"""
+
+        if self.num_donation == 0:
+            return 0
+        else:
+            return (self.total_donation/self.num_donation)
+
+    def add_donation(self, *args):
+        """add donation amount to a donor object"""
+        for i in args:
+            self.donations.append(i)
+
+    def __lt__(self, other):
+        """sorts the order"""
+        return self.total_donation < other.total_donation
+
+    def __repr__(self):
+        return self.name
+
+    def send_thank_you(self, money_amount):
+        """Sends a thank you note when a donation is made."""
+        return "Thank you {} for your donation amount of {} $. We thank you for your support".format(self.name, money_amount)
+
+
+class DonorCollection:
+
+    """
+    A class containing information of a list of Donor objects
+    """
+
+    def __init__(self, donors=[]):
+        """Create a collection of donor objects."""
+        self.donor_obs_list = donors
+
+    @property
+    def list_donors(self):
+        """Return a list of all donors in the collection."""
+        return self.donor_obs_list
+
+    @list_donors.setter
+    def list_donors(self, donors = []):
```

Comment:
don't use mutable objects for default keyword value, this can be really dangerous 

https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html